src/Microsoft.AspNetCore.Server.Kestrel/Internal/Http/Connection.cs

```diff
@@ -171,7 +171,7 @@ public virtual void OnSocketClosed()
         // Called on Libuv thread
         public void Tick(long timestamp)
         {
-            if (timestamp > Interlocked.Read(ref _timeoutTimestamp))
+            if (timestamp > SystemClock.VolatileRead(ref _timeoutTimestamp))
```

Comment:
Can we move it to some other class? Usages outside `SystemClock` look very weird.